Objects/ComingObjects/ComingObject.gd

```diff
@@ -5,14 +5,16 @@ signal timeout
 func _ready():
 	connect("timeout",self,"_on_timeout")
 	
-	var elf_pos = get_node("/root/World").find_node("Elf").global_position
-	var diff = global_position.x - elf_pos.x
+	var cam_pos = get_node("/root/World").find_node("Camera2D").position
+	print(cam_pos.x)
+	print(position.x)
+	var diff = get_viewport_rect().size.x
 	$ComingTimer.wait_time = diff / BackgroundData.move_speed
 	$ComingTimer.start()
 
-
-func _physics_process(delta):
-	position.x -= BackgroundData.move_speed * delta
+func _physics_process(delta): #unnecesary since camera2D was added
+	#position.x += BackgroundData.move_speed * delta * 2
+	pass
```

Comment:
'pass' is not good in this place. This doing nothing

Reply:
Yeah, I will delete this function.